packages/appium/lib/appium.js

```diff
@@ -410,6 +675,19 @@ class AppiumDriver extends DriverCore {
         );
         await driverInstance.updateSettings(jwpSettings);
       }
+
+      // if the user has asked for bidi support, and the driver says that it supports bidi,
+      // send our bidi url back to the user. The inner driver will need to have already saved any
+      // internal bidi urls it might want to proxy to, cause we are going to overwrite that
+      // information here!
+      if (dCaps.webSocketUrl && driverInstance.doesSupportBidi) {
+        const {address, port, basePath} = this.args;
+        const bidiUrl = `ws://${address}:${port}${basePath}${BIDI_BASE_PATH}/${innerSessionId}`;
```

Comment:
the protocol might not be `ws` if the server uses SSL. Check https://github.com/mykola-mokhnach/appium-devtools-plugin/blob/b8857ba5335645c1064d8cedb32898399b4d19a0/lib/mixins/proxy.js#L392 for more details.